### ISSUE TITLE
how to use username and password to  connnect es?

### ISSUE THREAD

**Issue by wqk151** [olivere/elastic#1253]
Please use the following questions as a guideline to help me answer
your issue/question without further inquiry. Thank you.

### Which version of Elastic are you using?

[ ] elastic.v7 (for Elasticsearch 7.x)
I use Elasticsearch 7.4

### Please describe the expected behavior
I want to  use username and password to connet my es

### Please describe the actual behavior
I use the api(elastic.v7) can not conn the es server.

### Any steps to reproduce the behavior?
my code is:

client, err = elastic.NewClient(
    // elastic.SetSniff(false),
    elastic.SetHealthcheckInterval(10*time.Second),
    elastic.SetMaxRetries(3),
    elastic.SetURL("http://xx:9200"),
    elastic.SetBasicAuth("username", "password"),
)

it not work,
report an error: "health check timeout: no Elasticsearch node available".

I didn't find any articles about using password authentication。
How can I solve this problem??
Please give me some tips,very thank you !!!

**Comment by zhousheng2** [comment]
client, err = elastic.NewClient(
elastic.SetSniff(false),
elastic.SetURL("http://xx:9200"),
elastic.SetBasicAuth("username", "password"),
)

**Comment by olivere** [comment]
This looks like a bug in that healthchecks don't use the username/password.

**Comment by andrewmichaelsmith** [comment]
Just to add a data point, I'm on ElasticSearch 7.5 and I also think I'm experiencing this issue.

The code I'm using is here: https://github.com/honeytrap/honeytrap/blob/master/pushers/elasticsearch/config.go

I've followed through #312 which largely seems to be people with misconfigured ES clusters. As you point out - if `https://localhost:9200/_nodes/http` returns IPs that are not accessible then sniffing won't work.

I've confirmed that the `publish_address` as accessible from the container my application is running in and tried turning sniff on/off. I am using password authentication.

In both cases I'm receiving:

```
health check timeout: no Elasticsearch node available
```

**Comment by andrewmichaelsmith** [comment]
Update - I've tested with my last known good set up (7.2.0) and I'm getting the same error, so it's possible I'm doing something wrong.

This application worked previously with 7.2.0, however I am possibly on a later version of the application (I didn't pin the version).

For reference, full log is:

```
DEBU 003 Using URL: https://es:9200 with index: index
DEBU 004 Using authentication with username: user and password.
CRIT 005 Error initializing channel es(elasticsearch): health check timeout: no Elasticsearch node available
```

**Comment by zhousheng2** [comment]
I can connect to es through user name and password. My code is as follows:
`// Create a client
	olCliente, err := elastic.NewClient(
		elastic.SetURL(host),
		elastic.SetSniff(false),
		elastic.SetBasicAuth(userName, password),
		//elastic.SetHealthcheckInterval(10*time.Second),
		elastic.SetErrorLog(log.New(os.Stderr, "ELASTIC ", log.LstdFlags)),
		elastic.SetInfoLog(log.New(os.Stdout, "", log.LstdFlags)))
	fmt.Println("connect to es success!")`

**Comment by zhousheng2** [comment]
I can connect to es through user name and password. My code is as follows:
// Create a client
	olCliente, err := elastic.NewClient(
		elastic.SetURL(host),
		elastic.SetSniff(false),
		elastic.SetBasicAuth(userName, password),
		//elastic.SetHealthcheckInterval(10*time.Second),
		elastic.SetErrorLog(log.New(os.Stderr, "ELASTIC ", log.LstdFlags)),
		elastic.SetInfoLog(log.New(os.Stdout, "", log.LstdFlags)))
	fmt.Println("connect to es success!")

**Comment by zhousheng2** [comment]
I can connect to es through user name and password. My code is as follows，Version 7.4 is used:
`
  // Create a client
	client, err := elastic.NewClient(
		elastic.SetURL(host),
		elastic.SetSniff(false),
		elastic.SetBasicAuth(userName, password),
		//elastic.SetHealthcheckInterval(10*time.Second),
		elastic.SetErrorLog(log.New(os.Stderr, "ELASTIC ", log.LstdFlags)),
		elastic.SetInfoLog(log.New(os.Stdout, "", log.LstdFlags)))
	fmt.Println("connect to es success!")

`

**Comment by olivere** [comment]
Hmm... I've looked over this and can't see a problem. Maybe it was due to #1179 that will be fixed in the upcoming release (see 9b52bb8bffc5859f647ecc12dacedf6f95dac564).

I will close this now. Feel free to re-open and I'll open it again. A runnable test case would be very helpful to reproduce the problem.

**Comment by feixiaohuijava** [comment]
i find the reason why cannot connect es.
you need do elastic.SetSniff(false)
